runtime/protocol/http/common/test/aws/smithy/kotlin/runtime/http/interceptors/FlexibleChecksumsResponseInterceptorTest.kt

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.http.interceptors
+
+import aws.smithy.kotlin.runtime.client.ExecutionContext
+import aws.smithy.kotlin.runtime.http.*
+import aws.smithy.kotlin.runtime.http.engine.HttpClientEngineBase
+import aws.smithy.kotlin.runtime.http.interceptors.FlexibleChecksumsResponseInterceptor.Companion.ChecksumHeaderValidated
+import aws.smithy.kotlin.runtime.http.operation.*
+import aws.smithy.kotlin.runtime.http.request.HttpRequest
+import aws.smithy.kotlin.runtime.http.request.HttpRequestBuilder
+import aws.smithy.kotlin.runtime.http.response.HttpCall
+import aws.smithy.kotlin.runtime.http.response.HttpResponse
+import aws.smithy.kotlin.runtime.io.SdkSource
+import aws.smithy.kotlin.runtime.io.source
+import aws.smithy.kotlin.runtime.time.Instant
+import aws.smithy.kotlin.runtime.util.get
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.CsvSource
+import kotlin.test.*
+
+data class TestInput(val value: String)
+data class TestOutput(val body: HttpBody)
+
+inline fun <reified I> newTestOperation(serialized: HttpRequestBuilder): SdkHttpOperation<I, TestOutput> =
```

Comment:
fix

This looks [duplicated](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/protocol/http/common/test/aws/smithy/kotlin/runtime/http/operation/TestOperation.kt#L19). Can we just import this or move it to a common location within the HTTP test package?

Reply:
It's not entirely duplicated, this one is customized to return `response.body` in a `TestOutput` type but the original one just returned a pre-determined `O` type. I needed access to the `response.body` in the output so I could consume the whole body and get the checksum validated, but I can't pass the body in using a pre-determined `O` because it needs to get wrapped internally in the interceptors / middleware.